learn/work/fairness-readmission/index.qmd

```diff
@@ -0,0 +1,517 @@
+
+```{r}
+#| label: "prop-by-race"
+#| message: false
+#| warning: false
+readmission %>%
+  # randomly split into groups
+  mutate(., group = sample(1:10, nrow(.), replace = TRUE)) %>%
+  group_by(race, group) %>%
+  # compute proportion readmitted by race + random `group`ing
+  summarize(prop = mean(readmitted == "Yes"), n = n()) %>%
+  # compute variation in the proportion by race.
+  # note that, by default, the output from above is grouped by race only.
+  summarize(mean = mean(prop), sd = sd(prop), n = sum(n))
+```
+
+The standard deviations `sd` are much larger for groups with a smaller number of observations `n`, even if the average proportions `mean` are similar. This variation will affect our analysis downstream in that fairness metrics measure variation across groups; the noise associated with proportions calculated for `race` groups with fewer values may overwhelm the signal associated with actual disparities in care for those groups.
+
+We have several options when considering how to address this:
+
+* **Remove rows arising from infrequent classes.** In this example, this would mean removing all rows with `race` values other than `"Caucasian"` or `"African American"`. This is the approach taken by other public analyses of these data, including the original study that this data arose from. An analysis resulting from this approach would ignore any disparities in care for `race` groups other than `"Caucasian"` or `"African American"`.
+* **Construct custom fairness metrics that address the increased variability associated with smaller counts.** We could scale the variability of the estimates for each `race` group before calculating fairness metrics.
+* **Bin the values for infrequent classes.** This would entail collapsing values of `race` other than `"Caucasian"` and `"African American"` into one factor level. This approach is somewhat of a hybrid of the two approaches above; we lose some granularity in information regarding care for `race` groups other than `"Caucasian"` and `"African American"`, but reduce the variability associated with the estimates for those groups in the process.
+
+For this analysis, we'll go with the last option.
+
+::: callout-note
+While we won't construct custom fairness metrics in this example, you can do so using the `new_groupwise_metric()` function in yardstick.
+:::
+
+Recoding that data column:
+
+```{r}
+#| label: "collapse-race"
+readmission_collapsed <-
+  readmission %>%
+  mutate(
+    race = case_when(
+      !race %in% c("Caucasian", "African American") ~ "Other",
```

Comment:
omg I know it's not needed but the `!race` part is needling me.

```suggestion
      !(race %in% c("Caucasian", "African American")) ~ "Other",
```

Reply:
Addressed in [115a87b](https://github.com/tidymodels/tidymodels.org/pull/52/commits/115a87bc8ac9c045789fa29970a5091889296d16). :)